lib/src/rules/missing_whitespace_between_adjacent_strings.dart

```diff
@@ -0,0 +1,113 @@
+
+Add a trailing whitespace to prevent missing whitespace between adjacent
+strings.
+
+With long text splitted accross adjacent strings it's easy to forget a
+whitespace between strings.
+
+**BAD:**
+```
+var s =
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed'
+  'do eiusmod tempor incididunt ut labore et dolore magna';
+```
+
+**GOOD:**
+```
+var s =
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed '
+  'do eiusmod tempor incididunt ut labore et dolore magna';
+```
+
+''';
+
+class MissingWhitespaceBetweenAdjacentStrings extends LintRule
+    implements NodeLintRule {
+  MissingWhitespaceBetweenAdjacentStrings()
+      : super(
+            name: 'missing_whitespace_between_adjacent_strings',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(NodeLintRegistry registry,
+      [LinterContext context]) {
+    final visitor = _Visitor(this);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends RecursiveAstVisitor<void> {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  @override
+  void visitAdjacentStrings(AdjacentStrings node) {
+    // skip regexp
+    if (node.parent is ArgumentList) {
+      final parentParent = node.parent.parent;
+      if (parentParent is InstanceCreationExpression &&
+              parentParent.staticElement.enclosingElement.name == 'RegExp' ||
+          parentParent is MethodInvocation &&
+              parentParent.realTarget == null &&
+              const ['RegExp', 'matches']
+                  .contains(parentParent.methodName.name)) {
+        return;
+      }
+    }
+
+    for (var i = 0; i < node.strings.length - 1; i++) {
+      final current = node.strings[i];
+      final next = node.strings[i + 1];
+      if (_visit(current, (l) => _endsWithWhitespace(l.last)) ||
+          _visit(next, (l) => _startsWithWhitespace(l.first))) {
+        continue;
+      }
+      if (!_visit(current, (l) => l.any(_hasWhitespace))) {
+        continue;
+      }
+      rule.reportLint(current);
+    }
+
+    return super.visitAdjacentStrings(node);
+  }
+
+  bool _visit(StringLiteral string, bool Function(Iterable<String>) test) {
+    if (string is SimpleStringLiteral) {
+      return test([string.value]);
+    } else if (string is StringInterpolation) {
+      return test(string.elements.map((e) {
+        if (e is InterpolationString) return e.value;
+        return '';
+      }));
+    } else if (string is AdjacentStrings) {
+      return string.strings.any((e) => _visit(e, test));
+    }
+    throw ArgumentError('${string.runtimeType}: $string');
+  }
+
+  bool _hasWhitespace(String value) =>
+      value.contains(eol) || value.contains(space);
+
+  bool _endsWithWhitespace(String string) =>
+      string.endsWith(eol) || string.endsWith(space);
+  bool _startsWithWhitespace(String string) =>
+      string.startsWith(eol) || string.startsWith(space);
```

Comment:
These seem a bit limited. Should we include both carriage return and newline? What about tab?

Reply:
Added `\r` and `\t`